### DISCUSSION TITLE
nvim-cmp suggestion for links with spaces

### DISCUSSION THREAD

**Discussion by vlcfaria** [epwalsh/obsidian.nvim#527] [Q&A] [answered]
Hello! I am relatively new to the nvim environment and I'm running into an issue where links generated by nvim-cmp are not correct, since they replace space characters with "-"s and everything is converted to lowercase. I have only run into this issue with headers, **the actual file ID reference inserts the spaces correctly**.

Example (the top link would be the correct one, which I wrote directly on the obsidian app):
![image](https://github.com/epwalsh/obsidian.nvim/assets/63602430/ece71a86-fd71-4e85-8f4b-914b76c3585c)

The file actually is found and correctly previewed on nvim, but the link does not work when viewing in the obsidian app. Am I missing out on a config of this plugin or nvim-cmp? For reference, I am using NvChad as a base config and my config for obsidian.nvim is:

```lua
{
    "epwalsh/obsidian.nvim",
    version = "*",
    ft = "markdown",
    dependencies = {
      "nvim-lua/plenary.nvim",
    },
    opts = {
      workspaces = {
        {
          name = "uni",
          path = "~/vaults/uni",
        },

        {
          name = "personal",
          path = "~/vaults/personal",
        },
      },
      note_id_func = function(title)
        -- Create note IDs in a Zettelkasten format with a timestamp and a suffix.
        -- In this case a note with the title 'My new note' will be given an ID that looks
        -- like '1657296016-my-new-note', and therefore the file name '1657296016-my-new-note.md'
        local suffix = ""
        if title ~= nil then
          -- If title is given, transform it into valid file name.
          suffix = title:gsub(" ", "-"):gsub("[^A-Za-z0-9-]", ""):lower()
        else
          -- If title is nil, just add 4 random uppercase letters to the suffix.
          for _ = 1, 4 do
          suffix = suffix .. string.char(math.random(65, 90))
          end
        end
        return tostring(os.time()) .. "-" .. suffix
      end,
      completion = {
        nvim_cmp = true,
        min_chars = 2,
      }
    },
  }
```
Thank you!!

**Comment by epwalsh** [comment, marked as answer]
Hey @Articusss, when linking to a header, by default the header part of the link is converted to a proper anchor link. You can change that behavior by customizing the `wiki_link_func` in your obsidian.nvim config.

**Reply by vlcfaria** [reply]
This worked, thank you!
